### PR TITLE
make reroute controller in MapboxNavigation#setRerouteController nullable

### LINKED ISSUE #5957: MapboxNavigation#setRerouteController ambiguity
### Description

`MapboxNavigation` class has 2 methods:
```
fun setRerouteController(rerouteController: RerouteController)
```
and
```
fun setRerouteController(
    rerouteController: NavigationRerouteController? = defaultRerouteController
)
```
And the `NavigationRerouteController` interface extends `RerouteController` interface.
When I try to do smth like that:
```
val navigationRerouteController: NavigationRerouteController = ...
mapboxNavigation.setRerouteController(navigationRerouteController)
```
there's an error: the call is ambiguous.
There's a workaround for this issue: `navigationRerouteController` has to be of a nullable type. But it seems like the API is not supposed to be this way: it's a common case to create a controller of type `NavigationRerouteController` and pass it to the setter and now we can't do that.

### Suggestions on how it can be fixed:
We can remove the nullability from the argument in the second method:
```
fun setRerouteController(
    rerouteController: NavigationRerouteController? = defaultRerouteController
)
```
and create a new method called `clearRerouteController`.
Then the issue will be gone, however, it breaks API.

Another solution is to make the argument in the first method nullable:
```
fun setRerouteController(rerouteController: RerouteController?)
```
It also solves the issue and I don't see how it can break API but it would be nice for someone to double-check.

### PR DESCRIPTION
**Pull request by dzinad**
### Description
Fixes #5957. See the details in the issue description.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ Mapbox welcomes participation and contributions from everyone.
 
 #### Bug fixes and improvements
 - Fixed reroute request interruption when setting the `NavigationRerouteController` [#5950](https://github.com/mapbox/mapbox-navigation-android/pull/5950).
+- Made `rerouteController` argument in `MapboxNavigation#setRerouteController` nullable. Null can be passed to disable automatic rerouting [#5957](https://github.com/mapbox/mapbox-navigation-android/pull/5957).
```
Comment (Zayankovsky) [inline CHANGELOG.md:10]: There should be a link to this PR, not to the issue it fixes. Also dots should be placed before the link, but I see that not everyone respects this rule. Take a look at https://github.com/mapbox/navigation-sdks/blob/main/documentation/android-changelog-guidelines.md. 
```suggestion
- Fixed reroute request interruption when setting the `NavigationRerouteController`. [#5950](https://github.com/mapbox/mapbox-navigation-android/pull/5950)
- Made `rerouteController` argument in `MapboxNavigation#setRerouteController` nullable. Null can be passed to disable automatic rerouting. [#5977](https://github.com/mapbox/mapbox-navigation-android/pull/5977)
```

Comment (dzinad) [reply CHANGELOG.md:10]: > Also dots should be placed before the link, but I see that not everyone respects this rule.

If you're speaking about the previous line, that was me. :)

File: libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt [hunk 2]
```diff
@@ -1195,11 +1195,31 @@ internal class MapboxNavigationTest : MapboxNavigationBaseTest() {
         assertFalse(mapboxNavigation.getRerouteController() === oldRerouteController)
     }
 
+    @Test
```
Comment (Guardiola31337) [inline libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt:1198]: NIT

Normally is good to structure the tests following the triple A 👀 https://speakerdeck.com/guardiola31337/elegant-unit-testing-mobilization-2016?slide=40 http://wiki.c2.com/?ArrangeActAssert

```kotlin
@Test
fun ignoreEnhancedLocationUpdates() {
    // Arrange
    
    // Act
    
    // Assert
}
```

Note the empty lines between Arrange / Act and Act / Assert.

It's cosmetics, although improves readability IMO.

The same thing applies to some other tests.

**Review by Guardiola31337 — APPROVED** [review]
Left a minor NIT.

**Review by Zayankovsky — APPROVED** [review]

File: libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt [hunk 3]
```diff
@@ -1198,6 +1198,38 @@ internal class MapboxNavigationTest : MapboxNavigationBaseTest() {
...
+    }
+
+    @Test
+    fun `set null navigation reroute controller`() = coroutineRule.runBlockingTest {
+        val oldController: NavigationRerouteController = mockk(relaxed = true)
+        val newController: NavigationRerouteController? = null
+        createMapboxNavigation()
+        mapboxNavigation.setRerouteController(oldController)
+
+        mapboxNavigation.setRerouteController(newController)
```
Comment (RingerJK) [inline libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt:1228]: nit: actions together
```suggestion

        mapboxNavigation.setRerouteController(oldController)
        mapboxNavigation.setRerouteController(newController)
```

**Review by RingerJK — APPROVED** [review]
LGTM

File: libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt (hunk 3 above)
Comment (dzinad) [reply libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt:1228]: It's intentional. First `setRerouteController` invocation is more about preparing. I want to check what happens when I set another one. I's vague but I see it this way. Maybe I'm wrong. What do you usually do in these situations?

Comment (RingerJK) [reply libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt:1228]: good point actually to a discussion. don't see any against your view. We can keep it as initially set 👍

**Pull request merged**